File: login_app/lib/models/employee_model.dart
class Employee {
  final String? id;
  final int empId;
  final String name;
  final String email;
  final String? password;
  final String department;
  final String designation;
  final String? city;
  final String? gender;
  final String? image;

  Employee({
    this.id,
    required this.empId,
    required this.name,
    required this.email,
    this.password,
    required this.department,
    required this.designation,
    this.city,
    this.gender,
    this.image,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      id: json['_id'],
      empId: json['emp_id'] is int ? json['emp_id'] : int.parse(json['emp_id'].toString()),
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      department: json['department'] ?? '',
      designation: json['designation'] ?? '',
      city: json['city'],
      gender: json['gender'],
      image: json['image'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'emp_id': empId,
      'name': name,
      'email': email,
      if (password != null && password!.isNotEmpty) 'password': password,
      'department': department,
      'designation': designation,
      'city': city,
      'gender': gender,
      'image': image,
    };
  }
}

File: employee-backend/models/employee.model.js
const mongoose = require('mongoose');

const EmployeeSchema = new mongoose.Schema(
  {
    emp_id: {
      type: Number,
      required: true,
      unique: true,
    },
    name: {
      type: String,
      required: true,
      trim: true,
    },
    email: {
      type: String,
      required: true,
      unique: true,
      trim: true,
      lowercase: true,
    },
    password: {
      type: String,
      required: true,
    },
    department: {
      type: String,
      required: true,
      trim: true,
    },
    designation: {
      type: String,
      required: true,
      trim: true,
    },
    city: {
      type: String,
      default: null,
    },
    gender: {
      type: String,
      default: 'Male',
    },
    image: {
      type: String,
      default: "https://clipart-library.com/new_gallery/301-3016414_headshot-silhouette-png.png",
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Employee', EmployeeSchema);